tensorflow_model_optimization/python/core/clustering/keras/cluster.py

```diff
@@ -126,6 +126,10 @@ def cluster_weights(to_cluster,
         format(cluster_centroids_init))
 
   def _add_clustering_wrapper(layer):
+    if (isinstance(layer, keras.Model)):
+      return keras.models.clone_model(layer,
```

Comment:
Would be good to raise an exception if it's a subclassed model, since `clone_model` is not supported in that case.